### PR TITLE
refactor(document-provider): Document provider to admin portal

### PR DESCRIPTION
**Pull request by GunnlaugurG**
## What
- Moved Document Provider to the admin portal and changed the Guard to use admin API scope to autherize API and UI and removed feature flag

## Why

- To be able to remove the hardcoded NationalId's from the environment file that indicates who is authorised to access the portal

## Screenshots / Gifs

Attach Screenshots / Gifs to help reviewers understand the scope of the pull request

## Checklist:

- [x] I have performed a self-review of my own code
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] Formatting passes locally with my changes
- [ ] I have rebased against main before asking for a review

### CODE REVIEW

**Review by peturgq — APPROVED** [review]

**Review by disaerna — APPROVED** [review]
LGTM

File: libs/api/domains/document-provider/src/lib/admin/admin-document.provider.resolver.ts [hunk 1]
```diff
@@ -1,57 +1,43 @@
...
+  UpdateContactInput,
+  UpdateHelpdeskInput,
+  UpdateOrganisationInput,
+} from '../dto'
+import { logger } from '@island.is/logging'
 
 @UseGuards(IdsUserGuard, ScopesGuard)
 @Resolver()
-export class DocumentProviderResolver {
+export class AdminDocumentProviderResolver {
```
Comment (eirikurn) [inline libs/api/domains/document-provider/src/lib/admin/admin-document.provider.resolver.ts:26]: Nice cleanup on this side! 👍

File: libs/portals/admin/core/src/lib/paths.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+export enum AdminPortalCorePaths {
```
Comment (eirikurn) [inline libs/portals/admin/core/src/lib/paths.ts:1]: We don't want to maintain a global paths object like before. Unless you discussed something different in the office.

These path objects should be inside each module instead.

File: libs/portals/admin/document-provider/src/module.ts [hunk 3]
```diff
@@ -1,26 +1,28 @@
...
 import { DocumentProviderPaths } from './lib/paths'
 import { Features } from '@island.is/feature-flags'
 import { m } from './lib/messages'
 
 export const documentProviderModule: PortalModule = {
   name: m.rootName,
-  featureFlag: Features.servicePortalDocumentProviderModule,
-  enabled: ({ isCompany }) => !isCompany,
+  enabled: ({ userInfo }) => {
+    console.log(userInfo)
```
Comment (eirikurn) [inline libs/portals/admin/document-provider/src/module.ts:11]: Remove log?

File: libs/portals/admin/document-provider/src/module.ts [hunk 4]
```diff
@@ -1,26 +1,28 @@
...
+    return userInfo.scopes.includes(AdminPortalScope.documentProvider)
+  },
   routes: ({ userInfo }) => [
     {
       name: m.rootName,
       path: DocumentProviderPaths.DocumentProviderRoot,
-      enabled: userInfo.scopes.includes(ApiScope.internal),
+      enabled: userInfo.scopes.includes(AdminPortalScope.documentProvider),
```
Comment (eirikurn) [inline libs/portals/admin/document-provider/src/module.ts:18]: Can skip enabled in the routes and just have it on the module?

File: tsconfig.base.json [hunk 5]
```diff
@@ -582,10 +582,16 @@
       "@island.is/next-ids-auth": ["libs/next-ids-auth/src/index.ts"],
       "@island.is/nova-sms": ["libs/nova-sms/src/index.ts"],
       "@island.is/plausible": ["libs/plausible/src/index.ts"],
+      "@island.is/portals-admin-icelandic-names-registry": [
+        "libs/portals/admin/icelandic-names-registry/src/index.ts"
+      ],
```
Comment (eirikurn) [inline tsconfig.base.json:587]: Unrelated change?

File: libs/service-portal/sessions/tsconfig.lib.json [hunk 6]
```diff
@@ -1,12 +1,12 @@
 {
   "extends": "./tsconfig.json",
   "compilerOptions": {
-    "outDir": "../../../dist/out-tsc",
+    "outDir": "../../../../dist/out-tsc",
```
Comment (eirikurn) [inline libs/service-portal/sessions/tsconfig.lib.json:4]: Unrelated change?

**Review by eirikurn — CHANGES_REQUESTED** [review]
Great job! 🙌

File: apps/services/documents/src/app/modules/document-provider/controllers/changelog.controller.ts [hunk 7]
```diff
@@ -1,12 +1,12 @@
...
-import { ApiScope } from '@island.is/auth/scopes'
+import { AdminPortalScope } from '@island.is/auth/scopes'
 import { Controller, Get, Param, UseGuards } from '@nestjs/common'
 import { ApiHeader, ApiOkResponse, ApiTags } from '@nestjs/swagger'
 import { DocumentProviderService } from '../document-provider.service'
 import { Changelog } from '../models/changelog.model'
 
 @UseGuards(IdsUserGuard, ScopesGuard)
-@Scopes(ApiScope.internal)
+@Scopes(AdminPortalScope.documentProvider)
```
[the remark this answers is not in the document]
Comment (GunnlaugurG) [reply apps/services/documents/src/app/modules/document-provider/controllers/changelog.controller.ts:9]: After taking a second look I saw that CreateOrganisation in the OrganisationController is also being called directly from document-provider-onboarding.service so I added internal scope to that route, I could not find another reference to the controller, but this will require more tests once it is out on dev where all the services are up and running.

**Review by eirikurn — APPROVED** [review]
LGTM 👍 🙌
